### PR TITLE
Add AuthenticationDisplayName to HttpSys

### LINKED ISSUE #21281: HttpSys does not set DisplayName for Authentication Scheme Options
### Describe the bug
``IISOptions``, ``IISServerOptions`` both have a property to set the Displayname. This is commonly used (e.g. IdentityServer) to show/hide buttons for the respective login protocols. If it is null, you hide the button.
Unfortunately HttpSys (which is kinda useful for integration testing of automatic (Negotiate) logins) never sets the DisplayName.
https://github.com/dotnet/aspnetcore/blob/c4f230195fd7fe325798c18b8c5f7e30523e7ac7/src/Servers/HttpSys/src/MessagePump.cs#L52-L55

May I suggest adding an ``AuthenticationDisplayName`` property to ``HttpSysOptions`` just like IISOptions or IISServerOptions and using it in the method above?
Example:
https://github.com/dotnet/aspnetcore/blob/c4f230195fd7fe325798c18b8c5f7e30523e7ac7/src/Servers/IIS/IISIntegration/src/IISOptions.cs#L15-L18

Related:
https://github.com/dotnet/aspnetcore/issues/5876 is a previous bug related to this, but was (automatically?) closed without further comments.
https://github.com/dotnet/aspnetcore/issues/14882 about DisplayNames

Workaround (if someone else needs that):
1. Replace ``IAuthenticationSchemeProvider`` with custom implementation (which delegates everything to the normal ``AuthenticationSchemeProvider`` 
2. Handle the AddScheme method manually, like
```csharp
public void AddScheme(AuthenticationScheme scheme)
{
    if (scheme.Name == HttpSysDefaults.AuthenticationScheme)
    {
        scheme = new AuthenticationScheme(scheme.Name, scheme.Name, scheme.HandlerType); // HttpSys has no DisplayName for some reason.
    }

   _baseProvider.AddScheme(scheme);
}
```

### Further technical details
- ASP.NET Core version: 3.1.3 (incl 5.0 preview)
C:\WINDOWS\system32>dotnet --info
.NET Core SDK (gemäß "global.json"):
 Version:   5.0.100-preview.5.20224.12
 Commit:    a513655c3c

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-preview.5.20224.12\

Host (useful for support):
  Version: 5.0.0-preview.5.20224.4
  Commit:  64a95836ac

.NET SDKs installed:
  2.1.801 [C:\Program Files\dotnet\sdk]
  2.2.207 [C:\Program Files\dotnet\sdk]
  2.2.401 [C:\Program Files\dotnet\sdk]
  3.0.100 [C:\Program Files\dotnet\sdk]
  3.1.200 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]
  3.1.300-preview-015115 [C:\Program Files\dotnet\sdk]
  5.0.100-preview.5.20224.12 [C:\Program Files\dotnet\sdk]
- The IDE (VS / VS Code/ VS4Mac) you're running on, and it's version
Visual Studio 2019 16.5

### PR DESCRIPTION
**Pull request by Kahbazi**
Fixes #21281

### CODE REVIEW

File: src/Servers/HttpSys/src/AuthenticationManager.cs [hunk 1]
```diff
@@ -61,6 +61,11 @@ public bool AllowAnonymous
         /// </summary>
         public bool AutomaticAuthentication { get; set; } = true;
 
+        /// <summary> 
+        /// Sets the display name shown to users on login pages. The default is null. 
+        /// </summary> 
+        public string AuthenticationDisplayName { get; set; }
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/AuthenticationManager.cs:67]: Reference assemblies need updating.
https://github.com/dotnet/aspnetcore/blob/master/docs/ReferenceAssemblies.md

**Comment by Tratcher** [comment]
Wow, that was fast. Did you set up notifications for HelpWanted issues somehow?

**Comment by Kahbazi** [comment]
No, I it was a normal subscription. I was just online ... and act fast :grin:

File: src/Servers/HttpSys/src/AuthenticationManager.cs (hunk 1 above)
Comment (Kahbazi) [reply src/Servers/HttpSys/src/AuthenticationManager.cs:67]: I couldn't generate reference assembly with scripts, so I just edit the file manually. I hope it works.

Comment (Tratcher) [reply src/Servers/HttpSys/src/AuthenticationManager.cs:67]: What happened when you tried?

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
So close:
```
-        public bool AuthenticationDisplayName { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
+        public string AuthenticationDisplayName { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
 ```

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**